README.md

```diff
@@ -257,6 +257,105 @@ replacing the type converters with something else. These elements, in possible
 conjunction with elements from the Hierarchy API, are flexible enough to build
 anything you need.
 
+## Special Type Parsing and Casting
+
+We use the `cast` library for casting values read in from configurations into their go types. The `cast` library 
+falls back to JSON for complex types expressed as string values. Here are some examples of how we parse different types:
+
+**[]string:**
+For a given configuration
+```go
+type Config struct {
+    TheSlice *[]string
```

Comment:
task: This should be a non-pointer type for the slice. Using a pointer to a slice is an uncommon case and will look unfamiliar to most folks.

Reply:
task: Similarly, all of the examples should use non-pointer types.